src/components/data-management/SamplesTableCells.jsx

```diff
@@ -128,6 +158,13 @@ const UploadCell = (props) => {
         file={uploadDetailsModalData}
         visible={uploadDetailsModalVisible}
```

Comment:
instead of using `visible` and passing it to `UploadDetailsModal` as prop, you can simply not render `UploadDetailsModal` when `uploadDetailsModalVisible` is false